### PR TITLE
Update configuration-reference.md

### LINKED ISSUE #4110: Config reference: `jobs.parameters` missing
# Missing Information
`parameters` is missing in config reference under `jobs`.

# Already Looked
It's mentioned here: https://circleci.com/docs/2.0/reusing-config/#authoring-parameterized-jobs

# Potential Locations
https://circleci.com/docs/2.0/configuration-reference/#job_name

### PR DESCRIPTION
**Pull request by thomasr8**
Added "parameters" value to table in Jobs section. This fixes #4110 .

# Description
A brief description of the changes.

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

### CODE REVIEW

File: jekyll/_cci2/configuration-reference.md [hunk 1]
```diff
@@ -144,6 +144,7 @@ docker | Y <sup>(1)</sup> | List | Options for [docker executor](#docker)
 machine | Y <sup>(1)</sup> | Map | Options for [machine executor](#machine)
 macos | Y <sup>(1)</sup> | Map | Options for [macOS executor](#macos)
 shell | N | String | Shell to use for execution command in all steps. Can be overridden by `shell` in each step (default: See [Default Shell Options](#default-shell-options))
+parameters | Y | List | A list of parameters that need to be passed as part of the request when performing this operation
```
Comment (teesloane) [inline jekyll/_cci2/configuration-reference.md]: I don't think the `parameters` are a `required` key 🤔

Comment (thomasr8) [reply jekyll/_cci2/configuration-reference.md]: Fixed

File: jekyll/_cci2/configuration-reference.md [hunk 2]
```diff
@@ -144,6 +144,7 @@ docker | Y <sup>(1)</sup> | List | Options for [docker executor](#docker)
 machine | Y <sup>(1)</sup> | Map | Options for [machine executor](#machine)
 macos | Y <sup>(1)</sup> | Map | Options for [macOS executor](#macos)
 shell | N | String | Shell to use for execution command in all steps. Can be overridden by `shell` in each step (default: See [Default Shell Options](#default-shell-options))
+parameters | N | List | A list of parameters that need to be passed as part of the request when performing this operation
```
Comment (Dominik-K) [inline jekyll/_cci2/configuration-reference.md]: @thomasr8 It's a `Map` like the `parameters` fields in the other 2.1 root fields (`commands`, etc.). I guess it's based on the same code.

Comment (thomasr8) [reply jekyll/_cci2/configuration-reference.md]: Ahhhh....thanks for the correction. I will go ahead and make the change in the docs this week. Thank you so much for helping with this.

**Pull request merged**